File: test/fixtures/c-project/services.c
#include "services.h"
#include "utils.h"
#include <string.h>

int validate_user(const User* user) {
    if (!user) return 0;
    return validate_email(user->email) && strlen(user->name) > 0;
}

int save_user(const User* user) {
    if (!validate_user(user)) return -1;
    return 0;
}

static int internal_check(const User* user) {
    return user->id > 0;
}

File: test/fixtures/rust-project/src/services.rs
use crate::models::User;
use crate::utils::validate_email;

pub struct UserService {
    users: Vec<User>,
}

impl UserService {
    pub fn new() -> Self {
        UserService { users: vec![] }
    }

    pub fn get_all(&self) -> Vec<User> {
        self.users.clone()
    }

    pub fn add(&mut self, user: User) -> Result<(), String> {
        if !validate_email(&user.email) {
            return Err("Invalid email".to_string());
        }
        self.users.push(user);
        Ok(())
    }

    fn find_by_id(&self, id: u32) -> Option<&User> {
        self.users.iter().find(|u| u.id == id)
    }
}
